modules/svg/src/SkSVGFilter.cpp

```diff
@@ -25,10 +25,18 @@ bool SkSVGFilter::parseAndSetAttribute(const char* name, const char* value) {
                    "primitiveUnits", name, value));
 }
 
+// NON-SKIA-UPSTREAMED CHANGE
+void SkSVGFilter::applyProperties(SkSVGRenderContext* ctx) const { this->onPrepareToRender(ctx); }
+// END OF NON-SKIA-UPSTREAMED CHANGE
+
 sk_sp<SkImageFilter> SkSVGFilter::buildFilterDAG(const SkSVGRenderContext& ctx) const {
     sk_sp<SkImageFilter> filter;
     SkSVGFilterContext fctx(ctx.resolveOBBRect(fX, fY, fWidth, fHeight, fFilterUnits),
                             fPrimitiveUnits);
+    // NON-SKIA-UPSTREAMED CHANGE
+    SkSVGRenderContext lctx(ctx);
+    this->applyProperties(&lctx);
```

Comment:
If I'm understanding correctly: after doing this, `lctx` will have all the `filter` node's parent's attributes which were already on `ctx`, plus it will have the attributes owned by the `filter` node itself (e.g. `color-interpolation-filters`)?

Reply:
No, `ctx` doesn't have `filter` node's attributes, it only has the attributes of its ancestor nodes. `lctx` is the same when just created, but after we call `applyProperties()`, it will also have the attributes of the `filter` node. And `localCtx` (below) will have the attributes of ancestors + the `filter` node + the specific child node.

Reply:
Thanks for clarifying.